backend/src/test/java/com/benine/backend/database/MySQLDatabaseTest.java

```diff
@@ -110,18 +120,134 @@ public final void testGetAllPreset() throws SQLException {
+        verifyConnectionClosed();
+    }
+
+    @Test
+    public final void testDeleteCamera() throws SQLException {
+        database.connectToDatabaseServer();
+        database.resetDatabase();
+        database.addCamera(1, "ip");
+        database.deleteCamera(1);
+        database.closeConnection();
+        verifySQLStatementExecuted("DELETE FROM presets WHERE camera_ID = 1");
+        verifySQLStatementExecuted("DELETE FROM camera WHERE ID = 1");
+        verifyCommitted();
+        verifyAllResultSetsClosed();
+        verifyConnectionClosed();
+    }
+
+    @Test
+    public final void testUseDatabase() throws SQLException {
+        database.connectToDatabaseServer();
+        database.resetDatabase();
+        database.useDatabase();
+        database.closeConnection();
+        verifySQLStatementExecuted("USE presetsdatabase");
+        verifyCommitted();
+        verifyAllResultSetsClosed();
+    }
+
+    @Test
+    public final void testCheckDatabase() throws SQLException {
+        database.connectToDatabaseServer();
+        MockResultSet result = statementHandler.createResultSet();
+        statementHandler.prepareGlobalResultSet(result);
+        Assert.assertFalse(database.checkDatabase());
+        database.closeConnection();
+        verifyAllResultSetsClosed();
+        verifyConnectionClosed();
+    }
+
+    @Test
+    public final void testGetPresetsFromResultSet() throws SQLException {
+        MockResultSet result = statementHandler.createResultSet();
+        statementHandler.prepareGlobalResultSet(result);
+        database.connectToDatabaseServer();
+        result.addColumn("pan", new Object[]{1});
+        result.addColumn("tilt", new Object[]{1});
+        result.addColumn("zoom", new Object[]{1});
+        result.addColumn("focus", new Object[]{1});
+        result.addColumn("iris", new Object[]{1});
+        result.addColumn("autofocus", new Object[]{1});
+        result.addColumn("panspeed", new Object[]{1});
+        result.addColumn("tiltspeed", new Object[]{1});
+        result.addColumn("autoiris", new Object[]{1});
+        result.addColumn("camera_ID", new Object[]{1});
+        result.next();
+        Preset preset = database.getPresetsFromResultSet(result);
+        assertEquals(preset, new Preset(new Position(1,1),1,1,1,true,1,1,true,1));
+    }
+
+    @Test
+    public final void testGetFailedPresetsFromResultSet() {
+        MockResultSet result = statementHandler.createResultSet();
+        database.connectToDatabaseServer();
+        assertNull(database.getPresetsFromResultSet(result));
+    }
+
+    @Test
+    public final void testCheckNewCameras() throws CameraConnectionException {
+        Camera camera = mock(Camera.class);
+        when(camera.getMacAddress()).thenReturn("mas");
+        database.connectToDatabaseServer();
+        ArrayList<Camera> list = new ArrayList<Camera>();
+        ArrayList<String> macs = new ArrayList<String>();
+        list.add(camera);
```

Comment:
I think the code above this comment could benefit from an @Before method, to reduce the amount of code. Some smaller lines such as `database.connectToDatabaseServer()` also appear in other tests.